tests/test-sources/env.nix

```diff
@@ -0,0 +1,15 @@
+{
+  env-variables = {
+    env = {
+      FOO = "1";
+      BAR_PATH = "/home/me/.local/share/bar";
+      BAZ_MAX_COUNT = "1000";
+    };
+
```

Comment:
Let's add some tests for edge-cases, such as variable names or values that may need `lib.escapeShellArg`